bin/create-app-secret

```diff
@@ -1,17 +1,57 @@
 #!/bin/bash
-read -s -p "Enter the Discovery Django secret key: " discovery_django_secret_key
-echo
-if [ -z "${discovery_django_secret_key}" ]; then
-  echo "The Discovery Django secret key cannot be empty."
-  exit 1
-fi
-read -s -p "Re-enter the secret key: " discovery_django_secret_key2
-echo
-if [ ! "${discovery_django_secret_key}" = "${discovery_django_secret_key2}" ]; then
-  echo "Discovery Django secret keys do not match."
-  exit 1
-fi
 
-podman secret rm discovery-django-secret-key >/dev/null 2>&1
-printf "${discovery_django_secret_key}" | podman secret create discovery-django-secret-key -
-podman secret ls --filter name=discovery-django-secret-key
+function set_django_secret_key {
+  podman secret rm discovery-django-secret-key >/dev/null 2>&1
+  printf '%s' "$1" | podman secret create discovery-django-secret-key -
+  podman secret ls --filter name=discovery-django-secret-key
+}
+
+function generate_key {
+  local discovery_django_secret_key
+  discovery_django_secret_key=$(head -c 64 /dev/urandom | base64 | tr -dc 'a-zA-Z0-9')
+  set_django_secret_key "${discovery_django_secret_key}"
+}
+
+function prompt_key {
+  local discovery_django_secret_key
+  read -srp "Enter the Discovery Django secret key: " discovery_django_secret_key
+  echo
+  if [ -z "${discovery_django_secret_key}" ]; then
+    echo "The Discovery Django secret key cannot be empty."
+    exit 1
+  fi
+  read -srp "Re-enter the secret key: " discovery_django_secret_key2
+  echo
+  if [ ! "${discovery_django_secret_key}" = "${discovery_django_secret_key2}" ]; then
+    echo "Discovery Django secret keys do not match."
+    exit 1
+  fi
+  set_django_secret_key "${discovery_django_secret_key}"
+}
+
+function prompt_for_generation {
+  while true; do
+    read -rp "Automatically generate new Django secret key? [y/n]: " yn
+    case $yn in
+    [Yy]*)
+      generate_key
+      exit
+      ;;
+    [Nn]*)
+      prompt_key
+      exit
+      ;;
+    esac
+  done
+}
+
+while getopts "Yy" opt; do
+  case $opt in
+  [Yy])
```

Comment:
usage below around line 254 should be updated to mention this.

echo "  create-app-secret  [-y]      - Create a Discovery application secret"

Reply:
Adding `-y` to run `discovery-installer create-app-secret -y` doesn't actually pass that extra argument along here in `discovery-installer`. It's only accepted if you invoke the `create-app-secret` script _directly_. So, for now, that docstring is still _technically_ correct.

We could expose it here pretty easily, but the existing argument parsing is already so spread out (hundreds of lines between `shift`s 😱), I'd prefer to rewrite and centralize that logic for readability and flexibility. However, I'd also rather punt any such rewrite to `[unspecified future date]` when we evolve this code into something of a `discoveryctl` which will hopefully have a _much more robust_ arg parser, whether that's the `getopt` program or Python's `argparse` module. 😉